src/utils/ensure-base-exists.utils.ts

```diff
@@ -39,68 +39,49 @@ export const ensureBaseTestsExists = async ({
   base: string | null;
   context: Context;
   octokit: InstanceType<typeof GitHub>;
-}): Promise<{ currentBaseSha: string } | null> => {
+}): Promise<{ shaToCompareAgainst: string | null }> => {
   const logger = log.getLogger(METICULOUS_LOGGER_NAME);
 
   // Running missing tests on base is only supported for Pull Request events
   if (event.type !== "pull_request" || !base) {
-    return null;
+    return { shaToCompareAgainst: null };
   }
 
   const { owner, repo } = context.repo;
   const baseRef = event.payload.pull_request.base.ref;
 
-  const currentBaseSha = await getHeadCommitForRef({
-    owner,
-    repo,
-    ref: baseRef,
-    octokit,
-  });
-
-  logger.debug(JSON.stringify({ base, baseRef, currentBaseSha }, null, 2));
-  if (base !== currentBaseSha) {
-    const message = `Pull request event received ${base} as the base commit but ${baseRef} \
-is now pointing to ${currentBaseSha}. Will use ${currentBaseSha} for Meticulous tests.`;
-    logger.warn(message);
-    ghWarning(message);
-  }
+  logger.debug(JSON.stringify({ base, baseRef }, null, 2));
 
   const testRun = await getLatestTestRunResults({
     client: createClient({ apiToken }),
-    commitSha: currentBaseSha,
+    commitSha: base,
   });
 
   if (testRun != null) {
-    logger.log(
-      `Tests already exist for commit ${currentBaseSha} (${testRun.id})`
-    );
-    return { currentBaseSha };
+    logger.log(`Tests already exist for commit ${base} (${testRun.id})`);
+    return { shaToCompareAgainst: base };
   }
 
   const { workflowId } = await getCurrentWorkflowId({ context, octokit });
 
   logger.debug(
-    `Debug: ${JSON.stringify(
-      { owner, repo, base, baseRef, currentBaseSha },
-      null,
-      2
-    )}`
+    `Debug: ${JSON.stringify({ owner, repo, base, baseRef }, null, 2)}`
   );
 
   const workflowRun = await getOrStartNewWorkflowRun({
```

Comment:
TODO: Looks like it's only really possible to trigger this on the commit for the head of the branch. So I'll need to tweak this code a bit (bring back the `base !== currentBaseSha` thing)